docs/Toy/toy/rewrites/arith_float_to_int.py

```diff
@@ -0,0 +1,92 @@
+"""
+Temporary workaround to lack of float support in our pipeline.
+
+The rewrites in this file convert arith operations on floats to operations on ints and add
+casts between the two.
+"""
+
+from typing import cast
+
+from xdsl.dialects import arith
+from xdsl.dialects.builtin import (
+    AnyFloat,
+    FloatAttr,
+    IntegerAttr,
+    ModuleOp,
+    UnrealizedConversionCastOp,
+    i32,
+)
+from xdsl.ir.core import MLContext
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    GreedyRewritePatternApplier,
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+
+
+class CastConstantOp(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: arith.Constant, rewriter: PatternRewriter):
+        if isinstance(op.value, IntegerAttr):
+            return
+
+        assert isinstance(op.value, FloatAttr)
+
+        op_value = cast(FloatAttr[AnyFloat], op.value)
+
+        value = op.value.value.data
+        assert (
+            value.is_integer()
+        ), f"Only support integer values in arith.Constant, got {value}"
+        value = int(value)
+
+        rewriter.replace_matched_op(
+            [
+                constant := arith.Constant.from_int_and_width(value, 32),
+                UnrealizedConversionCastOp.get(constant.results, (op_value.type,)),
+            ]
+        )
```

Comment:
```suggestion
class CastConstantOp(RewritePattern):
    @op_type_rewrite_pattern
    def match_and_rewrite(self, op: arith.Constant, rewriter: PatternRewriter):
        match op.value:
            case IntegerAttr():
                return
            case FloatAttr():
                op_value = cast(FloatAttr[AnyFloat], op.value)
                value = op_value.value.data
                assert (
                    value.is_integer()
                ), f"Only support integer values in arith.Constant, got {value}"
                value = int(value)

                rewriter.replace_matched_op(
                    [
                        constant := arith.Constant.from_int_and_width(value, 32),
                        UnrealizedConversionCastOp.get(
                            constant.results, (op_value.type,)
                        ),
                    ]
                )
            case _:
                assert False


```

Reply:
I don'f feel that thisd is an improvement.early returns are good, it communicates that there is a "happy path" down to the end of the function, whereas a match communicates that either case is equally interesting, I feel that the former applies better here.